### ISSUE TITLE
[spirv] Deserializer failed with "invalid iterator access"

### ISSUE THREAD

**Issue by denis0x0D** [tensorflow/mlir#298]
Hi @antiagainst, I'm facing a fail for phi.mlir test after the https://github.com/tensorflow/mlir/commit/0a1828ef3125f1da581f03b3147018b1ccf13cc3

with error:
`mlir-translate: : Assertion `isHandleInSync() && "invalid iterator access!"' failed.`

> #12 0x000055fc15827630 (anonymous namespace)::ControlFlowStructurizer::structurizeImpl() /home/llvm-project/llvm/projects/mlir/lib/Dialect/SPIRV/Serialization/Deserializer.cpp:1879:27

seems like this should be reordered to avoid invalid iterator fail:
  ```
 blockMergeInfo.try_emplace(newHeader, newMerge, newContinue);
 blockMergeInfo.erase(it);
``` 

Thanks.

**Comment by antiagainst** [comment]
Oh, good catch! Could you create a pull request for it? Thanks! :)

**Comment by denis0x0D** [comment]
@antiagainst thanks for the answer, created https://github.com/tensorflow/mlir/pull/299

**Comment by antiagainst** [comment]
Fixed via #299.
